### ISSUE TITLE
Multisite entries are 404ing [3.7.37]: 

### ISSUE THREAD

**Issue by clsse** [craftcms/cms#10878]
### What happened?

I'm using a multisite set-up using subdirectories (eg. @web/multisite-path)

Locally, everything is working fine.

On production, all entries are 404ing.

### Craft CMS version

3.7.37

### PHP version

7.4

### Installed plugins and versions

-

**Comment by brandonkelly** [comment]
Have you hard-coded the `@web` alias in `config/general.php` using the [aliases](https://craftcms.com/docs/3.x/config/config-settings.html#aliases) config setting?

**Comment by clsse** [comment]
I have not @brandonkelly, am I supposed to? And if so, what should I set it to?

**Comment by brandonkelly** [comment]
If you’re going to use the `@web` alias, it’s recommended that you explicitly set its value, to avoid cache injection attacks. It also could help work around situations like this where Craft/Yii isn’t able to reliably determine what the actual base URL is, perhaps due to a server configuration bug.

You would set it to whatever base URL you are expecting `@web` to be replaced with when you say the base site URL is `@web/multisite-path`.

**Comment by clsse** [comment]
Sorry, yes, I did have that set up. 

```
	'aliases' => [
		'@web' => getenv('SITE_URL'),
	],
```

**Comment by brandonkelly** [comment]
And does your production environment actually define a `SITE_URL` environment variable? You can check by going to `/admin/utilities/php-info` and searching for `SITE_URL`.

**Comment by clsse** [comment]
Yes it does!

**Comment by brandonkelly** [comment]
Try changing `getenv('SITE_URL')` to `craft\helpers\App::env('SITE_URL')`. 

Also double-check that `SITE_URL` is set correctly – it should be an absolute URL, which begins with `https://` or `http://`.

Also, is your System Status set to be Live in **Settings** → **General**? If you can’t see the **Settings** section on production (because `allowAdminChanges` is disabled), go to **Utilities** → **Project Config** and search the **Loaded Project Config Data** output for `system:`, and see if `live` is set to `true` below it.

**Comment by clsse** [comment]
Thank you. That did not work, unfortunately. Again, I don't know how relevant that is, but it works correctly on my local environment, but not on the production environment.

(Site was/is set correctly and system was/is live).

**Comment by brandonkelly** [comment]
Do the 404 errors look styled like your site’s normal 404 errors?

**Comment by clsse** [comment]
Yes, it says something to the effect of `template not found: /my/slug/here`

**Comment by brandonkelly** [comment]
If you request a path that is nested within the base path of a site (e.g. `/multisite-path/foo/bar/`, does the error include the baes URI from your site URL (`multisite-path`)? Or does the failed template path begin with whatever comes after (`foo/bar`)?

**Comment by clsse** [comment]
It includes the base URI.

Say my nested site url is set as `@web/something/something` 

The error for page "foo" shows as `/something/something/foo` 404

**Comment by brandonkelly** [comment]
In that case, Craft isn’t identifying the requested site properly. Something about the site’s base URL isn’t matching the requested URL (as far as Craft can tell).

Not much else I can help with from here unfortunately. If you can email SSH + admin credentials over to support@craftcms.com we can help you look into it further.
